docs/libraries/libraries-with-cli.md

```diff
@@ -0,0 +1,591 @@
+#else
+        private readonly HttpClient _client = new HttpClient();
+#endif
+
+#if NET40
+        // .NET 4.0 does not have async/await
+        public string GetDotNetCount()
+        {
+            string url = "http://www.dotnetfoundation.org/";
+          
+            var uri = new Uri(url);
+            
+            string result = "";
+            
+            // Lock here to provide thread-safety.
+            lock(_locker)
+            {
+                result = _client.DownloadString(uri);
+            }
+            
+            int dotNetCount = Regex.Matches(result, ".NET").Count;
+            
+            return $"Dotnet Foundation mentions .NET {dotNetCount} times!";
+        }
+#else
+        // .NET 4.5+ can use async/await!
+        public async Task<string> GetDotNetCountAsync()
+        {
+            string url = "http://www.dotnetfoundation.org/";
+            
+            // HttpClient is thread-safe, so no need to explicitly lock here
+            var result = await _client.GetStringAsync(url);
+            
+            int dotNetCount = Regex.Matches(result, ".NET").Count;
+            
+            return $"dotnetfoundation.orgmentions .NET {dotNetCount} times in its HTML!";
+        }
+#endif
+    }
+```
+
+And that's it!
+
+### But What about Portable Class Libraries (PCL)?
+
+If you want to target a PCL, you'll need to do one more thing before using `#if` directives to conditionally compile different targets: **you'll need to add a compilation definition in your** `project.json` **file**.
+
+For example, if you wanted to target [PCL profile 328](http://embed.plnkr.co/03ck2dCtnJogBKHJ9EjY/preview) (.NET 4.0, Windows 8, Windows Phone Silverlight 8, Windows Phone 8.1, Silverlight 5.0), you may want to refer to it to as "PORTABLE328" when cross-compiling.  Simply add it to the `project.json` file as a `compilationOptions` attribute:
+
+```
+{
+    "frameworks":{
+        "dotnet55":{
+           "dependencies":{
+                "System.Runtime":"4.0.0-rc1-*",
+                "System.Net.Http":"4.0.0-rc1-*",
+                "System.Thread.Tasks":"4.0.0-rc1-*"
+            }
+        },
+        ".NETPortable,Version=v4.0,Profile=Profile328":{
+            "compilationOptions": {
+                "define": [ "PORTABLE328" ]
+            },
+            "frameworkAssemblies":{
+                "mscorlib":"",
+                "System":"",
+                "System.Core":"",
+                "System.Net"
+            }
+        }
+    }
+}
+
+```
+
+Now you can conditionally compile against that target:
+
+```csharp
+#if !PORTABLE328
+using System.Net.Http;
+using System.Threading.Tasks;
+// Potentially other namespaces which aren't compatible with Profile 328
+#endif
+```
+
+And that's it! Because `PORTABLE328` is now recognized by the compiler, the PCL Profile 328 `.dll` generated by a compiler will not include `System.Net.Http` or `System.Threading.Tasks`.
```

Comment:
I'd prefer we just talk about the resulting artifacts as libraries, not `.dll`. It is more clear that we are getting at.